### PR TITLE
Enable variable batch size tests

### PR DESCRIPTION
**Pull request by jantonguirao**
Signed-off-by: Joaquin Anton <janton@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- Enables and fixes variable batch size tests in CI.

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Add a explicit call to the qa script, to invoke the variable batch size tests*
     *Fix failing tests*
     *Small refactoring to improve the readability of the test logs (each testcase is printed with the name of the operator)*
 - Affected modules and functionalities:
     *Python tests only*
 - Key points relevant for the review:
     *NA*
 - Validation and testing:
     *NA*
 - Documentation (including examples):
     *NA*

**JIRA TASK**: *[DALI-2031]*

### CODE REVIEW

File: dali/test/python/test_dali_variable_batch_size.py [hunk 1]
```diff
@@ -359,37 +351,37 @@ def pipe(max_batch_size, input_data, device):
...
+        dummy = fn.external_source(source=input_data, cycle=False, device=device)
+        data = fn.constant(fdata=3.1415, shape=(10, 10), device=device)
+        pipe.set_outputs(data, dummy)
         return pipe
 
     check_pipeline(
         generate_data(31, 13, custom_shape_generator(2, 4), lo=1, hi=255, dtype=np.uint8),
         pipeline_fn=pipe)
-
+test_constant()
```
Comment (szalpal) [inline dali/test/python/test_dali_variable_batch_size.py:384]:
```suggestion
```

File: dali/test/python/test_dali_variable_batch_size.py [hunk 2]
```diff
@@ -567,6 +559,9 @@ def generate_decoders_data(data_dir, data_extension):
     # File reader won't work, so I need to load audio files into external_source manually
     fnames = test_utils.filter_files(data_dir, data_extension)
 
+    nfiles = len(fnames)
+    for i in range(len(fnames), 10): # At leat 10 elements
+        fnames.append(fnames[-1])
```
Comment (szalpal) [inline dali/test/python/test_dali_variable_batch_size.py:564]: Could you add `TODO: remove` here? I'll remove these 3 lines with my `more_audio_data` PR

File: dali/test/python/test_dali_variable_batch_size.py [hunk 3]
```diff
@@ -240,83 +238,77 @@ def test_external_source():
...
+    (fn.warp_affine, {'matrix': (.1, .9, 10, .8, -.2, -20)}),
 ]
 
-
 def test_ops_image_custom_args():
-    for op in ops_image_custom_args:
-        print("\nTesting ", op)
-        image_data_helper(**op)
+    for op, args in ops_image_custom_args:
+        yield image_data_helper, op, args
```
Comment (szalpal) [inline dali/test/python/test_dali_variable_batch_size.py:267]: 👍🏻

File: dali/test/python/test_dali_variable_batch_size.py [hunk 4]
```diff
@@ -240,83 +238,77 @@ def test_external_source():
 
 
 def test_ops_image_default_args():
+    args = {}
     for op in ops_image_default_args:
-        yield image_data_helper, op
-
+        yield image_data_helper, op, args
```
Comment (JanuszL) [inline dali/test/python/test_dali_variable_batch_size.py:243]:
```suggestion
        yield image_data_helper, op, {}
```
?

File: dali/test/python/test_dali_variable_batch_size.py [hunk 5]
```diff
@@ -359,37 +351,37 @@ def pipe(max_batch_size, input_data, device):
...
 
-
 def test_constant():
     def pipe(max_batch_size, input_data, device):
         pipe = Pipeline(batch_size=max_batch_size, num_threads=4, device_id=0)
-        shape = fn.external_source(source=input_data, cycle=False, device='cpu')
-        data = fn.constant(fdata=3.1415, shape=shape, device=device)
-        pipe.set_outputs(data)
+        # just to drive the variable batch size.
+        dummy = fn.external_source(source=input_data, cycle=False, device=device)
```
Comment (JanuszL) [inline dali/test/python/test_dali_variable_batch_size.py:376]:
```suggestion
        but_size_setter = fn.external_source(source=input_data, cycle=False, device=device)
```

**Review by JanuszL — APPROVED** [review]

File: dali/test/python/test_dali_variable_batch_size.py (hunk 1 above)
Comment (jantonguirao) [reply dali/test/python/test_dali_variable_batch_size.py:384]: Good catch.

File: dali/test/python/test_dali_variable_batch_size.py (hunk 2 above)
Comment (jantonguirao) [reply dali/test/python/test_dali_variable_batch_size.py:564]: OK, I'll do

**Review by szalpal — APPROVED** [review]

**Comment by jantonguirao** [comment]
!build

**Comment by jantonguirao** [comment]
!build

**Pull request merged**